ext/opcache/ZendAccelerator.c

```diff
@@ -3019,11 +3019,15 @@ static void accel_move_code_to_huge_pages(void)
 		long unsigned int  start, end, offset, inode;
 		char perm[5], dev[10], name[MAXPATHLEN];
 		int ret;
+		extern char *__progname;
+		char buffer[MAXPATHLEN];
 
-		while (1) {
-			ret = fscanf(f, "%lx-%lx %4s %lx %9s %lu %s\n", &start, &end, perm, &offset, dev, &inode, name);
+		while (fgets(buffer, MAXPATHLEN, f)) {
+			ret = sscanf(buffer, "%lx-%lx %4s %lx %9s %lu %s\n", &start, &end, perm, &offset, dev, &inode, name);
 			if (ret == 7) {
-				if (perm[0] == 'r' && perm[1] == '-' && perm[2] == 'x' && name[0] == '/') {
+				// try to find the php text segment and map it into huge pages
+				char *is_substring = strstr(name, __progname);
+				if (perm[0] == 'r' && perm[1] == '-' && perm[2] == 'x' && name[0] == '/' && is_substring != NULL) {
```

Comment:
I would rewrite this like
```c
if (ret >= 6) {
    if (ret > 6
     && perm[0] == 'r' && perm[1] == '-' && perm[2] == 'x' && name[0] == '/'
     && strstr(name, __progname)) {
```
- lines without "name" are going to be skipped and won't stop the search
- strstr() call is going to be performed only after the cheaper checks

Reply:
Done. Thank you for your kindness~